testdata/project-v2/main.go

```diff
@@ -20,7 +20,6 @@ import (
 	"flag"
 	"os"
 
-	corev1 "k8s.io/api/core/v1"
```

Comment:
You are changing the generated file. You also need to change the scaffolding.
Otherwise the golden test will fail.

Reply:
https://github.com/kubernetes-sigs/kubebuilder/blob/3202ae8d778fc10680abf29366604034db0d98d9/pkg/scaffold/v2/main.go#L127

Reply:
But there is no `corev1 "k8s.io/api/core/v1"` in the scaffolding. I thought it was introduced manually but it may be introduced by some other file. `generate_golden.sh` maybe?

Reply:
https://github.com/kubernetes-sigs/kubebuilder/blob/3202ae8d778fc10680abf29366604034db0d98d9/generate_golden.sh#L74

Is this the line to blame?

Reply:
This is generated by https://github.com/kubernetes-sigs/kubebuilder/blob/eb3d7967cae29ba8bcf6893f359621551523bdeb/pkg/scaffold/v2/main.go#L51-L110
We currently just blindly add the import and the scheme, though corev1 has already been added.
It doesn't cause any problem by adding a scheme twice.

Reply:
One way to fix this is using [`KnownTypes`](https://github.com/kubernetes/apimachinery/blob/6eb29fdf75dcc3a9e33ec30ef32b25c923789dd1/pkg/runtime/scheme.go#L216-L227) to check if a <group, version> already exists before update it in `(m *Main) Update(opts *MainUpdateOptions)`.

Reply:
The problem is that we do not have an scheme as it is scaffolding the scheme. I guess we could make a copy but that would add as dependancies `k8s.io/apimachinery` for `NewScheme` and `KnownTypes`, and `k8s.io/client-go` for `AddToScheme`. What do you think about this dependencies?

Reply:
Thanks for bringing this up.
IMO it doesn't worth to introduce dependencies on client-go for this reason.
We are always trying to avoid having dependencies on k8s repos, since kubebuilder CLI is just a scaffolding tool and we don't want to confuse users that one kubebuilder release only works with a specific k8s version(s).